AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs

```diff
@@ -45,8 +47,51 @@ public void ConfigureTentacle()
             _processRunner.Run(_tentaclePath, string.Format("service {0} --install --start --console", InstanceArg));
         }
 
+        private void HandleFailedRemoval()
+        {
+            if (File.Exists(Path.Combine(_tentacleInstallPath, "Tentacle.config")))
+            {
+                try
+                {
+                    Log.Information("Detected existing tentacle.config - possible abrupt restart. Deleting tentacle.config.");
+                    File.Delete(Path.Combine(_tentacleInstallPath, "Tentacle.config"));
+                    Log.Information("Successfully removed tentacle.config from role.");
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Failed to remove certificate from Octopus server. Manual intevention required.");
+                }
+
+                try
+                {
+                    Log.Information("Attempting to remove tentacle from Octopus server.");
+                    DeleteMachine();
+                    Log.Information("Successfully removed tentacle from Octopus server.");
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Failed to remove tentacle from Octopus server. Manual intevention required.");
+                }
+
+            }
+        }
+
         public void DeleteMachine()
         {
+            try
+            {
+                DeleteMachineInternal();

```

Comment:
Yeah - this needs to change. One immediate retry isn't cutting it.
I could make it all exponential back-off, but the next one tries 5 times with 3 seconds between each.